app/controllers/decks_controller.rb

```diff
@@ -0,0 +1,129 @@
+class DecksController < ApplicationController
+  before_action :set_deck, only: %i[ show edit edit_name update update_name destroy ]
+  #Added this because kept getting error with invalid authenticity token. 
+  #Suspect issue might be the use the app within the tool consumer window
+  skip_before_action :verify_authenticity_token, :only => [:create, :update, :destroy]
```

Comment:
Yes, the forgery protection doesn't work if you're rendering in an iframe on Canvas.